### PR TITLE
[WIP] llvm: update to 8.0.0.

### PR DESCRIPTION
**Pull request by Cogitri**
Please don't review this yet. LLVM8 was scheduled to be
release today - I guess it'll be release soon-ish. I'll
switch stuff over to LLVM8 in the process of getting this
working.

[ci skip]

### CODE REVIEW

File: srcpkgs/llvm/template [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Template file for 'llvm'
 pkgname=llvm
```
Comment (maxice8) [inline srcpkgs/llvm/template]: imo just have llvm8 provides="llvm-8.0.0_${revision}"

Comment (Cogitri) [reply srcpkgs/llvm/template]: Hum, didn't we say on LLVM6->7 that provides is rather messy and that a metapackage is nicer to use?

**Comment by Cogitri** [comment]
Alright, LLVM8 final will take some time, rc3 just got release. See https://bugs.llvm.org/show_bug.cgi?id=40331 for the blockers for LLVM8. Relevant for #9248

File: srcpkgs/beignet/template [hunk 2]
```diff
@@ -1,14 +1,14 @@
 # Template file for 'beignet'
 pkgname=beignet
 version=1.3.2
-revision=4
+revision=5
 wrksrc="Beignet-${version}-Source"
```
Comment (maxice8) [inline srcpkgs/beignet/template]: archs= go above wrksrc=

Comment (Cogitri) [reply srcpkgs/beignet/template]: Sure, @maxice8

File: srcpkgs/llvm7/template [hunk 3]
```diff
@@ -13,37 +13,22 @@ configure_args="
  -DLLVM_ENABLE_FFI=ON
  -DLLVM_BINUTILS_INCDIR=/usr/include"
 short_desc="Low Level Virtual Machine"
-maintainer="Juan RP <xtraeme@voidlinux.org>"
+maintainer="Rasmus Thomsen <rasmus.thomsen@protonmail.com>"
```
Comment (maxice8) [inline srcpkgs/llvm7/template]: needs to be revbumped

Comment (Cogitri) [reply srcpkgs/llvm7/template]: Ah, yes, that went lost during the rebasing, @maxice8

**Comment by Cogitri** [comment]
Alright, the last thing blocking this is GNOME-Builder, which needs libclang (which isn't versioned) and currently doesn't build anymore. The GNOME-Builder update itself is blocked on glib which is blocked on glib 2.60.0.

**Comment by m0rk** [comment]
Hi, I checked out your branch but it was complaining `=> ERROR: xbps-src: unexistent file: /void-packages/srcpkgs/libllvm8/template` because there was no symbolic link from `libllvm8` to `llvm8`. I neither could find it in the changed files, so I think it is actually missing?

**Comment by Cogitri** [comment]
Oh, I think that went missing on the last rebase, thanks for the heads-up :)

File: srcpkgs/llvm8/template [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+	# XXX fails to cross compile due to python
+	subpackages+=" lldb lldb-devel"
+fi
+subpackages+=" lld lld-devel"
+
+case "$XBPS_TARGET_MACHINE" in
+	armv5*) broken="Tries to assemble VFP instructions with armv5tel arch";;
+esac
+
+post_extract() {
```
Comment (jnbr) [inline srcpkgs/llvm8/template]: please use post_patch() instead

File: srcpkgs/llvm8/files/patches/cfe/cfe-004-add-musl-triples.patch [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+   static const char *const AArch64beTriples[] = {"aarch64_be-none-linux-gnu",
+                                                  "aarch64_be-linux-gnu"};
+@@ -1882,7 +1882,9 @@ void Generic_GCC::GCCInstallationDetector::AddDefaultGCCPrefixes(
+   static const char *const ARMHFTriples[] = {"arm-linux-gnueabihf",
+                                              "armv7hl-redhat-linux-gnueabi",
+                                              "armv6hl-suse-linux-gnueabi",
+-                                             "armv7hl-suse-linux-gnueabi"};
++                                             "armv7hl-suse-linux-gnueabi",
++                                             "arm-linux-musleabihf",
++                                             "armv7l-linux-musleabihf"};
```
Comment (jnbr) [inline srcpkgs/llvm8/files/patches/cfe/cfe-004-add-musl-triples.patch]: to fix the linking problem on armv7 we need `armv7l-linux-gnueabihf` in this list

Comment (Cogitri) [reply srcpkgs/llvm8/files/patches/cfe/cfe-004-add-musl-triples.patch]: Fixed, I can't seem to see any other missing triplets, or do you know any?

Comment (jnbr) [reply srcpkgs/llvm8/files/patches/cfe/cfe-004-add-musl-triples.patch]: Thank you. I thinks it's complete now.

**Comment by Cogitri** [comment]
The only thing that doesn't work is beignet, which can been dead for about 2 years now, how about just dropping it?

File: srcpkgs/llvm8/files/patches/cfe/cfe-004-add-musl-triples.patch (hunk 5 above)
Comment (jnbr) [reply srcpkgs/llvm8/files/patches/cfe/cfe-004-add-musl-triples.patch]: yes

**Comment by Cogitri** [comment]
Alright, I've marked beignet as broken. Upstream has been dead for rather long now, we can't use libLLVM7 with it (because it needs libclang which we don't version and we can't build libclangxlibllvm7) and the switch from LLVM7->LLVM8 doesn't seem trivial for it.

**Comment by Cogitri** [comment]
If you want to test LLVM-8 on x86_64 you can get binaries from https://exp.exqa.de

**Comment by Cogitri** [comment]
Alright, I've added the mips cross triplets and marked GNOME-Builder nocross for now. From what I can see this is good to go, but please do take a very close look at all of this :)

**Pull request closed without merging**